test/org/sagebionetworks/bridge/services/ConsentServiceImplMockTest.java

```diff
@@ -124,4 +150,159 @@ public void noActivityEventIfDaoFails() {
+        
+        when(accountDao.getAccount(study, user.getEmail())).thenReturn(acct);
+        doThrow(new BridgeServiceException("Something bad happend", 500)).when(userConsentDao)
+            .withdrawConsent(user.getHealthCode(), study, UNIX_TIMESTAMP);
+        
+        ArgumentCaptor<Account> captor = ArgumentCaptor.forClass(Account.class);
+        
+        try {
+            consentService.withdrawConsent(study, user, new Withdrawal("For reasons."), UNIX_TIMESTAMP);
+            fail("Should have thrown an exception");
+        } catch(BridgeServiceException e) {
+        }
+        verify(accountDao).getAccount(any(), any());
+        verify(accountDao, times(2)).updateAccount(any(), captor.capture());
+        verifyNoMoreInteractions(sendMailService);
+        
+        Account account = captor.getAllValues().get(1);
+        assertEquals(1, account.getConsentSignatures().size());
+        assertNotNull(account.getActiveConsentSignature());
+        assertNull(account.getConsentSignatures().get(0).getWithdrewOn());
+    }
+
+    public static class SimpleAccount implements Account {
+        private String username;
+        private String firstName;
+        private String lastName;
+        private String email;
+        private String healthId;
+        private StudyIdentifier studyId;
+        private List<ConsentSignature> signatures = Lists.newArrayList();
+        private Map<String,String> attributes = Maps.newHashMap();
+        private Set<Roles> roles = Sets.newHashSet();
+        @Override
+        public String getUsername() {
+            return username;
+        }
+        @Override
+        public void setUsername(String username) {
+            this.username = username;
+        }
+        @Override
+        public String getFirstName() {
+            return firstName;
+        }
+        @Override
+        public void setFirstName(String firstName) {
+            this.firstName = firstName;
+        }
+        @Override
+        public String getLastName() {
+            return lastName;
+        }
+        @Override
+        public void setLastName(String lastName) {
+            this.lastName = lastName;
+        }
+        @Override
+        public String getAttribute(String name) {
+            return attributes.get(name);
+        }
+        @Override
+        public void setAttribute(String name, String value) {
+            attributes.put(name, value);
+        }
+        @Override
+        public String getEmail() {
+            return email;
+        }
+        @Override
+        public void setEmail(String email) {
+            this.email = email;
+        }
+        @Override
+        public ConsentSignature getActiveConsentSignature() {
+            List<ConsentSignature> actives = signatures.stream()
+                    .filter(signature -> signature.getWithdrewOn() == null).collect(Collectors.toList());
+            return (actives.isEmpty()) ? null : actives.get(actives.size()-1);
+        }
```

Comment:
Minor: Since this has a default implementation now, you don't need to duplicate it here.